examples/pxScene2d/src/pxScene2d.cpp

```diff
@@ -411,36 +411,39 @@ void pxObject::createNewPromise()
 void pxObject::dispose()
 {
   //rtLogInfo(__FUNCTION__);
-  mIsDisposed = true;
-  vector<animation>::iterator it = mAnimations.begin();
-  for(;it != mAnimations.end();it++)
+  if (!mIsDisposed)
   {
-    if ((*it).promise)
-      (*it).promise.send("reject",this);
-  }
+    mIsDisposed = true;
+    vector<animation>::iterator it = mAnimations.begin();
+    for(;it != mAnimations.end();it++)
+    {
+      if ((*it).promise)
+        (*it).promise.send("reject",this);
+    }
 
-  rtValue nullValue;
-  mReady.send("reject",nullValue);
+    rtValue nullValue;
+    mReady.send("reject",nullValue);
 
-  mAnimations.clear();
-  mEmit->clearListeners();
-  for(vector<rtRef<pxObject> >::iterator it = mChildren.begin(); it != mChildren.end(); ++it)
-  {
-    (*it)->dispose();
-    (*it)->mParent = NULL;  // setParent mutates the mChildren collection
-  } 
-  mChildren.clear();
-  deleteSnapshot(mSnapshotRef); 
-  deleteSnapshot(mClipSnapshotRef);
-  deleteSnapshot(mDrawableSnapshotForMask);
-  deleteSnapshot(mMaskSnapshot);
-  mSnapshotRef = NULL;
-  mClipSnapshotRef = NULL;
-  mDrawableSnapshotForMask = NULL;
-  mMaskSnapshot = NULL;
+    mAnimations.clear();
+    mEmit->clearListeners();
+    for(vector<rtRef<pxObject> >::iterator it = mChildren.begin(); it != mChildren.end(); ++it)
+    {
+      (*it)->dispose();
+      (*it)->mParent = NULL;  // setParent mutates the mChildren collection
+    } 
+    mChildren.clear();
+    deleteSnapshot(mSnapshotRef); 
+    deleteSnapshot(mClipSnapshotRef);
+    deleteSnapshot(mDrawableSnapshotForMask);
+    deleteSnapshot(mMaskSnapshot);
```

Comment:
It seems like `deleteSnapshot()` is not really delete... given the reference is disposed of outside `deleteSnapshot()` with an explicit NULL assignment.   Perhaps renaming it to `clearSnapshot()` might be clearer.   Just a thought.